src/dfcx_scrapi/core/conversation.py

```diff
@@ -609,26 +685,32 @@ def run_intent_detection(
 
     def _unpack_match(self, df: pd.DataFrame):
         """ Unpacks a 'match' column into four component columns.
+        if a match column is None, then all four columns will be None.
 
         Args:
           df: dataframe containing a column named match of types.Match
 
         Returns:
           A copy of df with columns match_type, confidence, parameters_set,
             and detected_intent instead of match.
+
         """
-        df = (
-            df
-            .copy()
+        return (
```

Comment:
`_unpack_match` changes LGTM per our discussion.

Reply:
Hello, @kmaphoenix , I forgot to ask you about this in the last call. In line 653, I added # pylint: disable=W0212 to prevent from the linter failing because it uses the underscore "_" to get the match_type._name_; it fails because this considers as a protected variable.

Either we can specify the global variable like MATCH_TYPES = {0:"MATCH_TYPE_UNSPECIFIED", 1:"INTENT", 2:"DIRECT_INTENT",...} and then map it by using MATCH_TYPES.get(m.match_type)
https://github.com/googleapis/google-cloud-python/blob/main/packages/google-cloud-dialogflow-cx/google/cloud/dialogflowcx_v3beta1/types/session.py#L1767C1-L1788C12
    class MatchType(proto.Enum):
        r"""Type of a Match.

        Values:
            MATCH_TYPE_UNSPECIFIED (0):
                Not specified. Should never be used.
            INTENT (1):
                The query was matched to an intent.
            DIRECT_INTENT (2):
                The query directly triggered an intent.
            PARAMETER_FILLING (3):
                The query was used for parameter filling.
            NO_MATCH (4):
                No match was found for the query.
            NO_INPUT (5):
                Indicates an empty query.
            EVENT (6):
                The query directly triggered an event.
            PLAYBOOK (9):
                The query was handled by a
                [``Playbook``][google.cloud.dialogflow.cx.v3beta1.Playbook].
        """

OR we can just add pylint: disable=W0212 to prevent from failing. 

Let me know what you think! thank you!